### PR TITLE
Identity provider key caching behavior configurable

### PR DESCRIPTION
**Pull request by strehle**
Issue: In UAA 74.5.x the JWKS is not cached. With newer UAAs there is a cache which is hard coded to 10 min.
If you have a fast rotation key, then this could be a problem.

Solution: deactivate the cache for such IdPs with cacheJwks=false options

### CODE REVIEW

**Comment by strehle** [comment]
Alternative to https://github.com/cloudfoundry/uaa/pull/2915

File: model/src/main/java/org/cloudfoundry/identity/uaa/provider/AbstractExternalOAuthIdentityProviderDefinition.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public enum OAuthGroupMappingMode {
     private boolean performRpInitiatedLogout = true;
     @JsonInclude(JsonInclude.Include.NON_NULL)
     private String authMethod;
+    private boolean cacheJwks = true;
```
Comment (strehle) [inline model/src/main/java/org/cloudfoundry/identity/uaa/provider/AbstractExternalOAuthIdentityProviderDefinition.java:56]: @peterhaochen47 . The name of the attribute can be different, if you want or have a proposal for another name

Comment (peterhaochen47) [reply model/src/main/java/org/cloudfoundry/identity/uaa/provider/AbstractExternalOAuthIdentityProviderDefinition.java:56]: This works

**Comment by peterhaochen47** [comment]
Thanks! LGTM. I pushed a commit to edit the docs. Please review.

**Review by peterhaochen47 — APPROVED** [review]

**Pull request merged**